src/proxy.ts

```diff
@@ -24,6 +24,12 @@ const web = async (
   return reply;
 };
 
+export const withSecure = (req: FastifyRequest, prefix: string) => {
+  const isSecure =
+    req.protocol === "https" || process.env.FORCE_SECURE === "true";
+  return isSecure ? `${prefix}s` : prefix;
+};
```

Comment:
Instead of a method, would it be worth adding this in index.ts via a `decorateRequest`? I think you could do something like:

```ts
server.decorateRequest('secure', function () {
  return this.protocol === "https" || process.env.FORCE_SECURE === "true";
});
```

As I think `this` will be scoped to the given incoming request. Ideally you'd then want to add a module augmentation to add this to the request type in types/fastify.d.ts